From 441f2a5907ae0988d38531faf74deca2cdb312f6 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Thu, 26 Dec 2019 17:49:36 +0200
Subject: [PATCH 4/4] pci: accept pcie base class id = 0x0

spr2803 pcie base class id is 0x0; this patch removes the case where it
doesn't allocate resources for such a device

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 drivers/pci/setup-bus.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pci/setup-bus.c b/drivers/pci/setup-bus.c
index 79b1824e83b4..f8a3cbaa55e8 100644
--- a/drivers/pci/setup-bus.c
+++ b/drivers/pci/setup-bus.c
@@ -182,7 +182,7 @@ static void __dev_sort_resources(struct pci_dev *dev,
 	u16 class = dev->class >> 8;
 
 	/* Don't touch classless devices or host bridges or ioapics.  */
-	if (class == PCI_CLASS_NOT_DEFINED || class == PCI_CLASS_BRIDGE_HOST)
+	if (/*class == PCI_CLASS_NOT_DEFINED ||*/ class == PCI_CLASS_BRIDGE_HOST)
 		return;
 
 	/* Don't touch ioapic devices already enabled by firmware */
-- 
2.17.1

